package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -0,0 +1,384 @@
+===========================================================================
+
+:Author: Paul Smith and Mateusz Bieniek
+:Year: 2019
+:Copyright: GNU Public License v3
+
+This module provides methods to find and analyse hydrogen bonds in a Universe.
+
+The :class:`HydrogenBondAnalysis` class is a new version of the original :class:`HydrogenBondAnalysis`
+class from the module mod:`MDAnalysis.analysis.hbonds.hbond_analysis`, which itself was modeled after the `VMD
+HBONDS plugin`_.
+
+.. _`VMD HBONDS plugin`: http://www.ks.uiuc.edu/Research/vmd/plugins/hbonds/
+
+
+Options:
+  - *donors_sel* [None] : Atom selection for donors. If `None`, then will be identified via the topology.
+  - *hydrogens_sel* [None] : Atom selection for hydrogens. If `None`, then will be identified via charge and mass.
+  - *acceptors_sel* [None] : Atom selection for acceptors. If `None`, then will be identified via charge.
+  - *d_h_cutoff* (Å) [1.2] : Distance cutoff used for finding donor-hydrogen pairs
+  - *d_a_cutoff* (Å) [3.0] : Distance cutoff for hydrogen bonds. This cutoff refers to the D-A distance.
+  - *d_h_a_angle_cutoff* (degrees) [150] : D-H-A angle cutoff for hydrogen bonds.
+  - *update_selections* [True] : If true, will update atom selections at each frame.
+
+
+Output:
+  - *frame* : frame at which a hydrogen bond was found
+  - *donor id* : atom id of the hydrogen bond donor atom
+  - *hydrogen id* : atom id of the hydrogen bond hydrogen atom
+  - *acceptor id* : atom id of the hydrogen bond acceptor atom
+  - *distance* (Å): length of the hydrogen bond
+  - *angle* (degrees): angle of the hydrogen bond
+
+Hydrogen bond data are returned in a :class:`numpy.ndarray` on a "one line, one observation" basis
+and can be accessed via :attr:`HydrogenBondAnalysis.hbonds`::
+
+    results = [
+        [
+            <frame>,
+            <donor index (0-based)>,
+            <hydrogen index (0-based)>,
+            <acceptor index (0-based)>,
+            <distance>,
+            <angle>
+        ],
+        ...
+    ]
+
+Classes
+-------
+
+.. autoclass:: HydrogenBondAnalysis
+   :members:
+
+   .. attribute:: hbonds
+
+"""
+from __future__ import absolute_import, division
+
+import numpy as  np
+
+from .. import base
+from MDAnalysis.lib.distances import capped_distance, calc_angles
+
+
+class HydrogenBondAnalysis(base.AnalysisBase):
+    """
+    Perform an analysis of hydrogen bonds in a Universe.
+    """
+
+    def __init__(self, universe, donors_sel=None, hydrogens_sel=None, acceptors_sel=None,
+                 d_h_cutoff=1.2, d_a_cutoff=3.0, d_h_a_angle_cutoff=150, update_selections=True):
+        """Set up atom selections and geometric criteria for finding hydrogen bonds in a Universe.
+
+        Parameters
+        ----------
+        universe : Universe
+            MDAnalysis Universe object
+        donors_sel : str
+            Selection string for the hydrogen bond donor atoms
+        hydrogens_sel :  str
```

Comment:
If hydrogens are left empty and are guessed then donors should also be left to None so that donors are properly guessed.

Reply:
I've added this to the docs